### PR TITLE
ci: add Debian environment to CI workflow

### LINKED ISSUE #2417: Add a Debian environment to CI workflow
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

Since we use debian as base image now, we should add a Debian environment to CI workflow:

https://github.com/apache/kvrocks/blob/unstable/.github/workflows/kvrocks.yaml#L423-L428

### Solution

--

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

### PR DESCRIPTION
**Pull request by shikharvashistha**
This commit adds Debian image to the CI workflow, closes #2417

@PragmaTwice

### CODE REVIEW

**Comment by PragmaTwice** [comment]
I think we need to add a `Setup Debian` step into `steps`, just like `Setup OpenSUSE` or `Setup ArchLinux`, to do some dependency installing.

**Comment by aleksraiden** [comment]
And why Debian 11 instead of 12, our docker base images based on bookworm-slim image

**Comment by shikharvashistha** [comment]
Updated 👍, kindly have a look and suggest changes, if required 

Best
@shikharvashistha

File: .github/workflows/kvrocks.yaml [hunk 1]
```diff
@@ -448,6 +451,16 @@ jobs:
           update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 100
           echo "NPROC=$(nproc)" >> $GITHUB_ENV
 
+      - name: Setup Debian
+        if: ${{ startsWith(matrix.image, 'debian') }}
+        run: |
+          apt update
+          apt install -y gcc g++ make wget git autoconf automake python3 python3-pip curl tar gzip cmake
```
Comment (aleksraiden) [inline .github/workflows/kvrocks.yaml]: I think, will be better: apt install -y build-essential wget cmake libtool python3 libssl-dev python3 python3-pip

File: .github/workflows/kvrocks.yaml [hunk 2]
```diff
@@ -426,6 +426,9 @@ jobs:
           - name: ArchLinux
             image: archlinux:base
             compiler: gcc
+          - name: Debian 12
+            image: debian:bookworm-slim
```
Comment (aleksraiden) [inline .github/workflows/kvrocks.yaml:427]: Maybe use a base image: bookworm, not a slim version

File: .github/workflows/kvrocks.yaml (hunk 1 above)
Comment (aleksraiden) [reply .github/workflows/kvrocks.yaml]: build-essential is a virtual package, including make and g++ and gcc.

File: .github/workflows/kvrocks.yaml (hunk 2 above)
Comment (PragmaTwice) [reply .github/workflows/kvrocks.yaml:427]: I think it's fine to use the same image as our docker image to cover all cases for any problem in the docker image.

File: .github/workflows/kvrocks.yaml [hunk 3]
```diff
@@ -426,6 +426,9 @@ jobs:
...
+            compiler: gcc
 
     runs-on: ubuntu-22.04
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml:430]:
```suggestion
    runs-on: ubuntu-22.04
    shell: bash
```

File: .github/workflows/kvrocks.yaml (hunk 2 above)
Comment (shikharvashistha) [reply .github/workflows/kvrocks.yaml:430]: 👍

Comment (aleksraiden) [reply .github/workflows/kvrocks.yaml:427]: Ok, make sense

**Comment by PragmaTwice** [comment]
Seems CI is stucked due to unknown reason. Let me retrigger it.

File: .github/workflows/kvrocks.yaml [hunk 4]
```diff
@@ -426,8 +426,12 @@ jobs:
...
+    shell: bash
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
```

File: .github/workflows/kvrocks.yaml [hunk 5]
```diff
@@ -448,6 +451,16 @@ jobs:
...
 
+      - name: Setup Debian
+        if: ${{ startsWith(matrix.image, 'debian') }}
+        run: |
+            apt update
+            apt install -y build-essential wget cmake libtool python3 libssl-dev python3 python3-pip
+            wget https://golang.org/dl/go1.22.5.linux-amd64.tar.gz
+            tar -C /usr/local -xzf go1.22.5.linux-amd64.tar.gz
+            export PATH=$PATH:/usr/local/go/bin
+            echo "NPROC=$(nproc)" >> $GITHUB_ENV
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
          apt update
          apt install -y build-essential wget cmake libtool python3 libssl-dev python3 python3-pip
          wget https://golang.org/dl/go1.22.5.linux-amd64.tar.gz
          tar -C /usr/local -xzf go1.22.5.linux-amd64.tar.gz
          export PATH=$PATH:/usr/local/go/bin
          echo "NPROC=$(nproc)" >> $GITHUB_ENV
```

File: .github/workflows/kvrocks.yaml [hunk 6]
```diff
@@ -448,6 +451,16 @@ jobs:
           update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 100
           echo "NPROC=$(nproc)" >> $GITHUB_ENV
 
+      - name: Setup Debian
+        if: ${{ startsWith(matrix.image, 'debian') }}
+        run: |
+          apt update
+          apt install -y build-essential wget cmake libtool python3 libssl-dev python3 python3-pip
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
          apt install -y build-essential wget cmake libtool python3 libssl-dev python3 python3-pip bash
```

File: .github/workflows/kvrocks.yaml [hunk 7]
```diff
@@ -448,6 +451,16 @@ jobs:
           update-alternatives --install /usr/bin/g++ g++ /usr/bin/g++-11 100
           echo "NPROC=$(nproc)" >> $GITHUB_ENV
 
+      - name: Setup Debian
+        if: ${{ startsWith(matrix.image, 'debian') }}
+        run: |
+          apt update
+          apt install -y build-essential wget cmake libtool python3 libssl-dev python3 python3-pip bash
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
          apt install -y build-essential wget cmake libtool python3 libssl-dev python3 python3-pip bash curl
```

File: .github/workflows/kvrocks.yaml [hunk 8]
```diff
@@ -495,7 +508,7 @@ jobs:
           ./x.py test go build $GOCASE_RUN_ARGS
 
       - name: Install redis-py
-        if: ${{ !startsWith(matrix.image, 'archlinux') }}  # already installed
+        if: ${{ !startsWith(matrix.image, 'archlinux') }}
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
        if: ${{ !startsWith(matrix.image, 'archlinux') }}
        shell: bash
```

File: .github/workflows/kvrocks.yaml [hunk 9]
```diff
@@ -495,7 +508,8 @@ jobs:
...
+        shell: bash
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
```

File: .github/workflows/kvrocks.yaml [hunk 10]
```diff
@@ -495,7 +509,7 @@ jobs:
           ./x.py test go build $GOCASE_RUN_ARGS
 
       - name: Install redis-py
-        if: ${{ !startsWith(matrix.image, 'archlinux') }}  # already installed
+        if: ${{ !startsWith(matrix.image, 'archlinux') }}
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]:
```suggestion
        if: ${{ !startsWith(matrix.image, 'archlinux') }}  # already installed
```

File: .github/workflows/kvrocks.yaml [hunk 11]
```diff
@@ -308,9 +308,17 @@ jobs:
...
+        run: |
+          python3 -m venv venv
+          source venv/bin/activate
+          
+          
       - name: Install redis-py
-        run: pip3 install redis==4.3.6
+        run: |
+          source venv/bin/activate
+          pip3 install redis==4.3.6
```
Comment (PragmaTwice) [inline .github/workflows/kvrocks.yaml]: Please don't change unrelated part : )

Comment (shikharvashistha) [reply .github/workflows/kvrocks.yaml]: I've updated according to the error logs in the last CI run, where it demands for a virtual environment before package installation, can you kindly check and suggest changes

Comment (PragmaTwice) [reply .github/workflows/kvrocks.yaml]: You need to change `build-and-test-in-container` instead of `build-and-test`.

**Review by Sandra-Amina-Boss — APPROVED** [review]

**Review by Sandra-Amina-Boss — APPROVED** [review]

**Review by Sandra-Amina-Boss — APPROVED** [review]

**Review by Sandra-Amina-Boss — APPROVED** [review]

**Review by Sandra-Amina-Boss — APPROVED** [review]

**Comment by PragmaTwice** [comment]
I will close this PR as there is no further progress. Feel free to reopen it.

**Pull request closed without merging**